research/delf/delf/python/training/README.md

```diff
@@ -118,11 +118,116 @@ files can take up to 12 hours and up to 500 GB of space disk.*
+```
+curl -Os http://storage.googleapis.com/delf/resnet50_imagenet_weights.tar.gz
+tar -xzvf resnet50_imagenet_weights.tar.gz
+```
 
+Assuming the TFRecord files were generated in the `gldv2_dataset/tfrecord/` directory, running 
+the following command should start training a model and output the results in the `gldv2_training`
+directory:
 ```
-python3 train.py \
+python train.py \
   --train_file_pattern=gldv2_dataset/tfrecord/train* \
   --validation_file_pattern=gldv2_dataset/tfrecord/validation*
+  --imagenet_checkpoint=resnet50_weights_tf_dim_ordering_tf_kernels_notop.h5 \
+  --dataset_version=gld_v2_clean \
+  --logdir=gldv2_training/
+```
+
+On a multi-GPU machine the batch size can be increased to speed up the training using the `--batch_size` parameter. On a 8 Tesla P100 GPUs machine you can set the batch size to `256`:
+```
+--batch_size=256
+```
+
+## Exporting the Trained Model
+
+Assuming the training output, the TensorFlow checkpoint, is in the `gldv2_training` directory,
+running the following command exports the model in the `gldv2_model` directory:
+```
+python model/export_model.py \
+  --ckpt_path=gs://dananghel-delf/training/internal/delf_weights \
+  --export_path=gldv2_model
 ```
+
+## Testing the Trained Model
+
+After the trained model has been exported, it can be used to extract DELF features from 2 images 
+of the same landmark and to perform a matching test between the 2 images based on the extracted 
+features to validate they represent the same landmark.
+
+Start by downloading the Oxford buildings dataset:
+````
+mkdir data && cd data
+wget http://www.robots.ox.ac.uk/~vgg/data/oxbuildings/oxbuild_images.tgz
+mkdir oxford5k_images oxford5k_features
+tar -xvzf oxbuild_images.tgz -C oxford5k_images/
+cd ../
+echo data/oxford5k_images/hertford_000056.jpg >> list_images.txt
+echo data/oxford5k_images/oxford_000317.jpg >> list_images.txt
+````
+
+Make a copy of the [`delf_config_example.pbtxt`](../examples/delf_config_example.pbtxt) 
+protobuffer file which configures the DELF feature extraction. Update the file by making the
+following changes:
+* set the `model_path` attribute to the directory containing the exported model, `gldv2_model`
+  in this example
+* add at the root level the attribute `is_tf2_exported` with the value `true`
+* set to `false` the `use_pca` attribute inside `delf_local_config`
+
+The ensuing file should resemble the following:
+```
+model_path: "gldv2_model"
+image_scales: .25
+image_scales: .3536
+image_scales: .5
+image_scales: .7071
+image_scales: 1.0
+image_scales: 1.4142
+image_scales: 2.0
+is_tf2_exported: true
+delf_local_config {
+  use_pca: false
+  # Note that for the exported model provided as an example, layer_name and
+  # iou_threshold are hard-coded in the checkpoint. So, the layer_name and
+  # iou_threshold variables here have no effect on the provided
+  # extract_features.py script.
+  layer_name: "resnet_v1_50/block3"
+  iou_threshold: 1.0
```

Comment:
I think these lines can be omitted

Reply:
Done, I removed them.

Reply:
Reran the feature extraction and image matching test with the new *.pbtxt file and it works.

model_path: "gldv2_model"
image_scales: .25
image_scales: .3536
image_scales: .5
image_scales: .7071
image_scales: 1.0
image_scales: 1.4142
image_scales: 2.0
is_tf2_exported: true
delf_local_config {
  use_pca: false
  max_feature_num: 1000
  score_threshold: 100.0
}